util/db/dbutil.go

```diff
@@ -83,28 +87,38 @@ func makeAccessorImpl(dbfilename string, readOnly bool, inMemory bool, params []
 	// Solution is to create a connection using a safe synchronization barrier right here.
 	// The connection goes to a connection pool inside Go's sql package and will be re-used when needed.
 	// See https://github.com/algorand/go-algorand/issues/846 for more details.
-	var err error
-	db.Handle, err = sql.Open("sqlite3", URI(dbfilename, readOnly, inMemory)+"&"+strings.Join(params, "&"))
-
-	if err == nil {
-		// create a connection to safely initialize SQLite once
-		initFn := func() {
-			var conn *sql.Conn
-			if conn, err = db.Handle.Conn(context.Background()); err != nil {
-				db.Close()
-				return
-			}
-			if err = conn.Close(); err != nil {
-				db.Close()
-			}
-		}
-		sqliteInitOnce.Do(initFn)
-		if err != nil {
-			// init failed, db closed and err is set
-			return db, err
-		}
-		err = db.runInitStatements()
+
+	// we create a dummy in-memory database and close it right after. This would suffice to initialize the sqlite library threading logic.
+	var dbHandler *sql.DB
+	dbHandler, sqliteConnections.initilizationError = sql.Open("sqlite3", URI("algorand-sqlite3-database-initializer", false, true)+"&_journal_mode=wal")
+	if sqliteConnections.initilizationError != nil {
+		return
+	}
+	sqliteConnections.initilizationError = dbHandler.Ping()
+	if sqliteConnections.initilizationError != nil {
+		return
+	}
+	sqliteConnections.initilizationError = dbHandler.Close()
```

Comment:
are you sure Close does not leak into sqlite3_uninitialize() somehow?

Reply:
change was discarded.